http-body-util/src/combinators/frame.rs

```diff
@@ -13,6 +13,7 @@ impl<'a, T: Body + Unpin + ?Sized> Future for Frame<'a, T> {
     type Output = Option<Result<http_body::Frame<T::Data>, T::Error>>;
 
     fn poll(mut self: Pin<&mut Self>, ctx: &mut task::Context<'_>) -> task::Poll<Self::Output> {
+        let _ = Pin::new(&mut self.0).poll_progress(ctx)?;
```

Comment:
Here's an example of the slightly silly way this ends up being used to preserve a `Poll<Result<_, _>>` return type.